VTExts.md

```diff
@@ -0,0 +1,387 @@
+    | ---------- | ---------- | --------------------------------------------------- |
+    | Format ID  | `uint32_t` | The format to check (e.g., `1` for CF_TEXT).        |
+-   **Out Stack:**
+    | Argument   | Type     | Description                                         |
+    | ---------- | -------- | --------------------------------------------------- |
+    | Available  | `int8_t` | `1` if available, `0` if not.                         |
+
+##### `FARTTY_INTERACT_CLIP_SETDATA` ('s')
+Puts data onto the clipboard.
+
+-   **In Stack:**
+    | Argument   | Type        | Description                             |
+    | ---------- | ----------- | --------------------------------------- |
+    | Format ID  | `uint32_t`  | The ID of the data format.              |
+    | Data       | `raw bytes` | The data buffer. Its size is determined by the next argument. |
+    | Data Size  | `uint32_t`  | The size of the data buffer.            |
+-   **Out Stack:**
+    | Argument   | Type       | Description                                  |
+    | ---------- | ---------- | -------------------------------------------- |
+    | Status     | `int8_t`   | `1` for success, `0` for failure, `-1` if not open. |
+    | Data ID    | `uint64_t` | (Optional) A unique ID for the data (e.g., CRC64). |
+
+##### `FARTTY_INTERACT_CLIP_GETDATA` ('g')
+Retrieves data from the clipboard.
+
+-   **In Stack:**
+    | Argument   | Type       | Description                  |
+    | ---------- | ---------- | ---------------------------- |
+    | Format ID  | `uint32_t` | The format to retrieve.      |
+-   **Out Stack:**
+    | Argument   | Type        | Description                                  |
+    | ---------- | ----------- | -------------------------------------------- |
+    | Data Size  | `uint32_t`  | The size of the retrieved data. `0` on failure, `-1` if not open. |
+    | Data       | `raw bytes` | The data buffer.                             |
+    | Data ID    | `uint64_t`  | (Optional) A unique ID for the data.         |
+
+---
+
+### `FARTTY_INTERACT_IMAGE` ('i')
+
+Namespace for all image rendering operations.
+
+-   **In Stack:**
+    | Argument     | Type   | Description                                         |
+    | ------------ | ------ | --------------------------------------------------- |
+    | Sub-command  | `char` | A character identifying the specific image action.    |
+    | ...          | ...    | Arguments specific to the sub-command.              |
+-   **Out Stack:** Varies by sub-command.
+
+#### Image Operations
+
+##### `FARTTY_INTERACT_IMAGE_CAPS` ('c')
+Queries the terminal's image rendering capabilities.
+
+-   **In Stack:** None.
+-   **Out Stack:**
+    | Argument          | Type       | Description                                  |
+    | ----------------- | ---------- | -------------------------------------------- |
+    | Capabilities      | `uint64_t` | A bitmask of `WP_IMGCAP_*` flags (e.g., `0x02` for RGBA). |
```

Comment:
RGBA is not 2, here re currently defines capabilities:
#define WP_IMGCAP_RGBA      0x01 // supports WP_IMG_RGB/WP_IMG_RGBA
#define WP_IMGCAP_SCROLL    0x02 // supports existing image scrolling
#define WP_IMGCAP_ROTATE    0x04 // supports existing image rotation